src/domain/challenge/base-challenge/base.challenge.service.ts

```diff
@@ -289,32 +321,48 @@ export class BaseChallengeService {
     return challenge.lifecycle;
   }
 
-  async getMembersCount(
+  public async getMembersCount(
     baseChallenge: IBaseChallenge,
     repository: Repository<BaseChallenge>
   ): Promise<number> {
     const community = await this.getCommunity(baseChallenge.id, repository);
     return await this.communityService.getMembersCount(community);
   }
 
-  async getAspectsCount(
+  public async getAspectsCount(
     baseChallenge: IBaseChallenge,
     repository: Repository<BaseChallenge>
   ): Promise<number> {
-    const context = await this.getContext(baseChallenge.id, repository);
-    const canvasesCount = await this.aspectsService.getAspectsInContextCount(
-      context.id
+    const collaboration = await this.getCollaboration(
+      baseChallenge.id,
+      repository
+    );
+    const callouts = await this.collaborationService.getCalloutsOnCollaboration(
+      collaboration
+    );
+    // TODO: fix after full callouts added, all aspects are currently under one callout
+    const defaultCallout = callouts[0];
+    const canvasesCount = await this.aspectsService.getAspectsInCalloutCount(
+      defaultCallout.id
     );
     return canvasesCount;
   }
 
-  async getCanvasesCount(
+  public async getCanvasesCount(
     baseChallenge: IBaseChallenge,
     repository: Repository<BaseChallenge>
   ): Promise<number> {
-    const context = await this.getContext(baseChallenge.id, repository);
-    const canvasesCount = await this.canvasService.getCanvasesInContextCount(
-      context.id
+    const collaboration = await this.getCollaboration(
+      baseChallenge.id,
+      repository
+    );
+    const callouts = await this.collaborationService.getCalloutsOnCollaboration(
+      collaboration
+    );
+    // TODO: fix after full callouts added, all aspects are currently under one callout
+    const defaultCallout = callouts[0];
```

Comment:
Just iterate over all callouts and sum it up, no need to default it. After the migration, the query to the callouts will simply return one callout, the default one.